src/main/java/fr/zcraft/zlib/components/gui/ArrayPromptGui.java

```diff
@@ -0,0 +1,93 @@
+package fr.zcraft.zlib.components.gui;
+
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import fr.zcraft.zlib.components.gui.ExplorerGui;
+import fr.zcraft.zlib.components.gui.Gui;
+import fr.zcraft.zlib.tools.Callback;
+
+/* Built from recommendations in PR https://github.com/zDevelopers/zLib/pull/14 */
+public class ArrayPromptGui<T> extends ExplorerGui<T> {
+
+	private Callback<T> cb;
+	private String purpose;
+	private T[] datas;
+	private ItemRenderer<T> renderer;
+	private boolean closeOnChoice;
+
+	/**
+	 * @param callback
+	 *            Callback called when the player made a choice
+	 * @param player
+	 *            The player making the choice
+	 * @param purpose
+	 *            The purpose (gui's title)
+	 * @param datas
+	 *            An array of datas to display
+	 * @param renderer
+	 *            Interface for building gui's ItemStack from array values
+	 * @param closeOnChoice
+	 *            Close the interface when the player has choosen if true
+	 */
+	public <A> ArrayPromptGui(Callback<T> callback, Player player, String purpose, T[] datas, ItemRenderer<T> renderer,
```

Comment:
I'm not sure about this. You're still using callbacks in the constructor instead of a real subclass. For the callback, I agree (cc @prokopyl thought on that?), but as for the item renderer, grmpf.

I suggest you to make the `ArrayPromptGui` class abstract, and to add a new abstract method:
```java
abstract public ItemStack getViewItem(T data);
```
(renamed following the previously established convention in the [explorer GUI](https://github.com/zDevelopers/zLib/blob/master/src/main/java/fr/zcraft/zlib/components/gui/ExplorerGui.java#L459-L465)).

So, instead of using it like:
```java
new ArrayPromptGui((object) -> doSth(object), player, "title", data, new ItemRenderer<T>()
{
    @Override
    public ItemStack build(T data) { /* ... */ }
}, true);
```
you will instead use it like this:
```java
new ArrayPromptGui((object) -> doSth(object), player, "title", data, true)
{
    @Override
    public ItemStack getViewItem(T data) { /* ... */ }
};
```

It's not really different, and the advantage is you can create a concrete child class for repeated use instead of uglying storing the rendered in a variable or, even worse, duplicating some code. Also, child classes could override the constructor to set some defaults :) .

Exactly like in the [`PaginatedTextView`](https://github.com/zDevelopers/zLib/blob/master/src/main/java/fr/zcraft/zlib/tools/commands/PaginatedTextView.java#L200-L288) ;) .